pkg/metadata/host/host.go

```diff
@@ -249,21 +250,47 @@ func getMeta() *Meta {
 	return m
 }
 
-func getContainerMeta() map[string]string {
+func getContainerMeta(timeout time.Duration) map[string]string {
+	wg := sync.WaitGroup{}
 	containerMeta := make(map[string]string)
+	// protecting the above map from concurrent access
+	mutex := &sync.Mutex{}
 
 	for provider, getMeta := range container.DefaultCatalog {
-		meta, err := getMeta()
-		if err != nil {
-			log.Debugf("Unable to get %s metadata: %s", provider, err)
-			continue
-		}
-		for k, v := range meta {
-			containerMeta[k] = v
+		wg.Add(1)
+		go func(provider string, getMeta container.MetadataProvider) {
+			defer wg.Done()
+			meta, err := getMeta()
+			if err != nil {
+				log.Debugf("Unable to get %s metadata: %s", provider, err)
+				return
+			}
+			mutex.Lock()
+			for k, v := range meta {
+				containerMeta[k] = v
```

Comment:
Should we check that one MetadataProvider is not overwriting values from another one ? Could it happened ?

Reply:
It technically could but should not 😬